flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/RuntimeEnvironment.java

```diff
@@ -345,7 +353,12 @@ public ExecutorService getAsyncOperationsThreadPool() {
     }
 
     @Override
-    public ThroughputCalculator getThroughputMeter() {
-        return throughputCalculator;
+    public void setCheckpointStorage(CheckpointStorage checkpointStorage) {
+        this.checkpointStorage = checkpointStorage;
+    }
```

Comment:
Could we add a check here to prevent resetting the storage?
And also to other recently added setters?